controller/integration/test/parallel/parallel_test.go

```diff
@@ -0,0 +1,224 @@
+// +build k8srequired
+
+package parallel
+
+import (
+	"testing"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/cenkalti/backoff"
+	"github.com/giantswarm/apiextensions/pkg/apis/core/v1alpha1"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/operatorkit/controller"
+	"github.com/giantswarm/operatorkit/controller/integration/testresource"
+	"github.com/giantswarm/operatorkit/controller/integration/wrapper"
+	"github.com/giantswarm/operatorkit/controller/integration/wrapper/nodeconfig"
+)
+
+// Test_Finalizer_Integration_Parallel is a integration test to
+// check that multiple controllers can function in parallel.
+func Test_Finalizer_Integration_Parallel(t *testing.T) {
+	var err error
+	objName := "test-obj"
+	testNamespace := "finalizer-integration-parallel-test"
+
+	testFinalizerA := "operatorkit.giantswarm.io/test-operator-a"
+	operatorNameA := "test-operator-a"
+
+	testFinalizerB := "operatorkit.giantswarm.io/test-operator-b"
+	operatorNameB := "test-operator-b"
+
+	// We create the first controller "A" here with its own resource.
+	var trA *testresource.Resource
+	{
+		c := testresource.Config{}
+
+		trA, err = testresource.New(c)
+		if err != nil {
+			t.Fatal("expected", nil, "got", err)
+		}
+	}
+
+	testWrapperA, err := setupController(testNamespace, operatorNameA, trA)
+	if err != nil {
+		t.Fatal("expected", nil, "got", err)
+	}
+
+	controllerA := testWrapperA.Controller()
+
+	// We create the second controller "B" and give it a different resource.
+	var trB *testresource.Resource
+	{
+		c := testresource.Config{}
+
+		trB, err = testresource.New(c)
+		if err != nil {
+			t.Fatal("expected", nil, "got", err)
+		}
+	}
+
+	testWrapperB, err := setupController(testNamespace, operatorNameB, trB)
+	if err != nil {
+		t.Fatal("expected", nil, "got", err)
+	}
+
+	controllerB := testWrapperB.Controller()
+
+	// We setup the namespace in which we test. We use the wrapper of controller A
+	// here, it makes no difference if we use the wrapper of A or B.
+	testWrapperA.MustSetup(testNamespace)
+	defer testWrapperA.MustTeardown(testNamespace)
+
+	// We start the controllers.
+	go controllerA.Boot()
+	go controllerB.Boot()
+
+	// We create an object without any finalizers.
+	// Creation is retried because the existance of a CRD might have to be ensured.
+	obj := &v1alpha1.NodeConfig{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      objName,
+			Namespace: testNamespace,
+		},
+	}
+	var createdObj interface{}
+	operation := func() error {
+		createdObj, err = testWrapperA.CreateObject(testNamespace, obj)
```

Comment:
The testwrapper here uses the k8sclient or g8sclient internally, it doesn't matter if we use `A` or `B` as both are configured the same way. I stick with `A` throughout this test for consistency.